drivers/tapdisk-image.c

```diff
@@ -287,6 +289,53 @@ tapdisk_image_close_chain(struct list_head *list)
 		tapdisk_image_close(image);
 }
 
+static int
+find_parent_nbd_path(int prt_devnum, char **prt_nbd_path)
+{
+	int err;
+	char *pattern;
+	glob_t glbuf = { 0 };
+
+	/* This is glob pattern not a regex */
+	err = asprintf(&pattern, "%s/nbd[0-9]*.%d", BLKTAP2_CONTROL_DIR, prt_devnum);
+	if (err == -1) {
+		errno =-errno;
+		return err;
+	}
+
+	err = glob(pattern, 0, NULL, &glbuf);
+	switch (err) {
+	case GLOB_NOMATCH:
+		EPRINTF("%s, failed to find parent NBD path", pattern);
+		err = -ENOENT;
+		goto done;
+
+	case GLOB_ABORTED:
+	case GLOB_NOSPACE:
+		err = -errno;
+		EPRINTF("%s: glob failed, err %d", pattern, err);
+		goto done;
+	}
+
+	if (glbuf.gl_pathc != 1) {
+		/* Got more than one match */
+		err = -EINVAL;
+		EPRINTF("%s got more than one match, %lu", pattern, glbuf.gl_pathc);
+		goto done;
+	}
+	*prt_nbd_path = strdup(glbuf.gl_pathv[0]);
+	err = 0;
+
+done:
+	if (glbuf.gl_pathv)
+		globfree(&glbuf);
+
+	if (pattern)
```

Comment:
I think this 'if' is redundant too, free is safe to call with a NULL pointer according to the manpage.